File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_l(string x);
int count_w(string x);
int count_s(string x);
float Compute_index(float x, float y, float z);
void print(int z);
int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    float letters = count_l(text);
    float words = count_w(text);
    float sentences = count_s(text);

    // Compute the Coleman-Liau index
    float index1 = Compute_index(letters, words, sentences);
    int index2 = round(index1);
    // Print the grade level
    print(index2);
}
int count_l(string x)
{
    int letters = 0;
    for (int i = 0, len = strlen(x); i < len; i++)
    {
        if (isalpha(x[i]) != 0)
        {
            letters++;
        }
    }
    return letters;
}
int count_w(string x)
{
    int words = 1;
    for (int i = 0, len = strlen(x); i < len; i++)
    {
        if (isblank(x[i]) != 0)
        {
            words++;
        }
    }
    return words;
}
int count_s(string x)
{
    int sentences = 0;
    for (int i = 0, len = strlen(x); i < len; i++)
    {
        if (x[i] == '.' || x[i] == '?' || x[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}
float Compute_index(float x, float y, float z)
{
    float index = 0;
    float L = (x / y) * 100;
    float S = (z / y) * 100;
    index = 0.0588 * L - 0.296 * S - 15.8;
    return index;
}

void print(int z)
{
    if (z >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (z < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", z);
    }
}

File: sentimental-readability/readability.py
import string
import re


# Coleman-Liau index is computed as
# 0.0588 * L - 0.296 * S - 15.8
# L :average number of letters per 100 words
# S :average number of sentences per 100 words
# L : count the number of spaces + 1
def get_string():
    s = input("Text:")
    return s


def main():
    text = get_string()
    wordlist = text.split()
    total_slist = re.findall(r".*?[?!.]", text)
    words = len(wordlist)
    total_s = len(total_slist)
    total_l = 0
    for char in text:
        if char.isalpha():
            total_l += 1
    Letters = float((total_l / words) * 100)
    Sentences = float((total_s / words) * 100)
    Grade = round(0.0588 * Letters - 0.296 * Sentences - 15.8)
    if Grade < 1:
        print("Before Grade 1")
    elif Grade >= 1 and Grade <= 16:
        print("Grade", Grade)
    elif Grade > 16:
        print("Grade 16+")


main()
